### ISSUE TITLE
integration with reactQL

### ISSUE THREAD

**Issue by shomanishikawa** [faceyspacey/react-universal-component#21]
Thank you for creating an awesome package!

Currently attempting to integrate with universal component workflow to the reactql boilerplate - https://reactql.org/

Trying to integrate the most basic implementation without webpack-flush-chunks (for now)

My home component looks like this:
```js
import React from 'react';
import universal from 'react-universal-component'

const UniversalComponent = universal(props => import(`./${props.page}`));

class Home extends React.Component {
  render() {
    return (
      <div>
      <UniversalComponent page={'Hero'} />
      </div>
    );
  }
}

export default Home;

```

My babel plugins for the client:
```js
             plugins: [
                'transform-object-rest-spread',
                'syntax-dynamic-import',
                'transform-regenerator',
                'transform-class-properties',
                'transform-decorators-legacy',
                'universal-import'
              ],
```

For the server:
```js
          plugins: [
            'transform-object-rest-spread',
            'syntax-dynamic-import',
            'transform-class-properties',
            'transform-decorators-legacy',
            'universal-import',
            'dynamic-import-webpack',
          ],
```

I also have the LimitChunkCountPlugin on the server webpack and the CommonsChunkPlugin set up on the client webpack.

When I load the SSR'ed version of the page, the source response shows 
`<div data-reactid="27">Loading...</div>`
where the component would be.

Once the client syncs however, that content is replaced with 
`Error: Cannot read property 'call' of undefined`
This obviously also brings up the checksum issue

The output in the js file is as follows:
```js
var UniversalComponent = __WEBPACK_IMPORTED_MODULE_4_react_universal_component___default()(function (props) {
  console.log(props.page);
  return __WEBPACK_IMPORTED_MODULE_2_babel_plugin_universal_import_universalImport_js___default()({
    id: './',
    file: '/Users/shomanishikawa/Desktop/app2/src/components/Home.jsx',
    load: function load() {
      return Promise.all([__webpack_require__(363)("./" + props.page), __WEBPACK_IMPORTED_MODULE_1_babel_plugin_universal_import_importCss_js___default()('' + props.page)]).then(function (proms) {
        return proms[0];
      });
    },
    path: function path() {
      return __WEBPACK_IMPORTED_MODULE_0_path___default.a.join(__dirname, './' + props.page);
    },
    resolve: function resolve() {
      return /*require.resolve*/(__webpack_require__(364).resolve("./" + props.page));
    },
    chunkName: function chunkName() {
      return '' + props.page;
    }
  });
});
```

Further, when I change the UniversalComponent definition to a hardcoded path:
`const UniversalComponent = universal(props => import(`./Hero`));`
Once the client syncs, it properly shows the Hero module content. This also gets rid of the checksum issue, because both the client and server first initialize in their 'Loading...' state. I guess this is partly due to the fact that it skips the universal-import?

Repo here: https://github.com/shomanishikawa/reactql-app2

Would really love to get this working, so I can continue on to implementing webpack-flush-chunks. I will be opening a PR into the reactql boilerplate if I can get this working, as this really seems like the ideal agnostic universal component option.

Thanks again!

**Comment by shomanishikawa** [comment]
I was able to fix this issue by upgrading to webpack ^3.5.0 but now I am running into the original issue I was experiencing, which is that flushChunksNames() returns an empty array, called after renderToString.

the components are both still initializing in the 'loading...' state on both client and server.

```js
    // Wait for GraphQL data to be available in our initial render,
    // before dumping HTML back to the client
    await getDataFromTree(components);

    // Full React HTML render
    const html = ReactDOMServer.renderToString(components);

    // this returns empty array [], even though we're hitting a dynamic import
    console.log(flushChunkNames());

```

my network tab also shows that I am loading the separate chunk dynamically on the client.

repo has been updated: https://github.com/shomanishikawa/reactql-app2

Thanks again!

**Comment by shomanishikawa** [comment]
Closing as this has become an issue with webpack-flush-chunks
